app/models/host_alias.rb

```diff
@@ -0,0 +1,13 @@
+class HostAlias < ActiveRecord::Base
+  belongs_to :domain
```

Comment:
I don't think so
you could have `mywebserver.example.com` been an alias to `api.foobar.com` and to `www.foobar.com`